docs/index.rst

```diff
@@ -8,7 +8,11 @@ Table of contents
 .. toctree::
    :maxdepth: 2
 
-   dev/index
+   overview/index
```

Comment:
The overview is really well-written, and since there is nothing more in that hierarchy, it makes more sense IMO to include it here instead.